### PR TITLE
adds daily run to offline mode

### PR DESCRIPTION
**Pull request by Nexllon**
Fixed fetchDailyRunSeed() to reject properly, had unhandled errors.
Made a function to handle daily generation inside initDailyRun() and added a isLocal check to run based on that:
online -> generate based on seed from fetchDailyRunSeed()
offline -> generated based on a simple base64 string date

### CODE REVIEW

File: src/phases.ts [hunk 1]
```diff
@@ -332,9 +332,17 @@ export class TitlePhase extends Phase {
           this.scene.sessionPlayTime = 0;
           this.end();
         });
-      }).catch(err => {
-        console.error("Failed to load daily run:\n", err);
-      });
+      };
+
+      if (!Utils.isLocal){
```
Comment (bennybroseph) [inline src/phases.ts:337]: Would like a very simple comment above this line explaining this addition

**Comment by Admiral-Billy** [comment]
I can confirm offline works as expected, cannot verify online although it *looks* fine to me heh

**Review by bennybroseph — CHANGES_REQUESTED** [review]

File: src/phases.ts (hunk 1 above)
Comment (Nexllon) [reply src/phases.ts:337]: Sure, just did!

**Review by bennybroseph — APPROVED** [review]

**Comment by Flashfyre** [comment]
Daily runs on offline mode bypass the once per day reward. This is a change in balance between online and offline.

**Comment by Nexllon** [comment]
> Daily runs on offline mode bypass the once per day reward. This is a change in balance between online and offline.

That's true, I was thinking about how easily moddable everything is and forgot about the normal players who want/need the same experience.
I'll try to make an offline check for that using local storage as the rest of the offline info

**Pull request closed without merging**